vortex-expr/src/binary.rs

```diff
@@ -50,8 +50,8 @@ impl VortexExpr for BinaryExpr {
             Operator::Lte => compare(lhs, rhs, ArrayOperator::Lte),
             Operator::Gt => compare(lhs, rhs, ArrayOperator::Gt),
             Operator::Gte => compare(lhs, rhs, ArrayOperator::Gte),
-            Operator::And => and(lhs, rhs),
-            Operator::Or => or(lhs, rhs),
+            Operator::And => and_kleene(lhs, rhs),
+            Operator::Or => or_kleene(lhs, rhs),
```

Comment:
I don't think we want this, we should have another operator for this or just separate expression

Reply:
HMMM, ok we do want this

Reply:
If we want SQL semantics, we want this.